File: cs214/labs/lab01/04/circle_area.rb
#! /usr/bin/ruby
# circle_area.rb computes the area of a circle given its radius
# 
# Input: the radius of a circle
# Precondition: the radius is not negative
# Output: the area of the circle.
#
# Begun by: Dr. Nelesen, for CS 214 at Calvin College.
# Completed by: Charles Blum
# Date: 06 February 2014
###############################################################

# function circleArea returns a circle's area, given its radius
# Parameters: r, a number
# Precondition: r > 0.
# Returns: the area of a circle whose radius is r.
PI = 3.1415927			#defines a constant PI
def circleArea(r)		#defines a function circleArea and expects a parameter r
    PI * r ** 2 		#calculates the area
end				#returns the value

if __FILE__ == $0		#eqv. to int main
   print "Enter the radius: "	#output to terminal for the user
   radius = gets.chomp.to_f	#gets the radius from the user
   print "Area is: "		#output to terminal for the user
   puts circleArea(radius)	#append the returned value from circleArea to the terminal
end				#end of int main

File: cs214/labs/lab01/01/circle_area.cpp
/* circle_area.cpp computes the area of a circle.
 *
 * Input: The radius of the circle.
 * Precondition: The radius is a positive number.
 * Output: The area of the circle.
 *
 * Begun by: Dr. Nelesen, for CS 214, at Calvin College.
 * Completed by: Charles Blum
 * Date: 06 February 2014
 **********************************************************/

#include <iostream>             // cin, cout, >>, <<
using namespace std;

/* function circleArea() computes a circle's area, given its radius.
 * Parameter: r, a double
 * Precondition: r is not negative.
 * Returns: the area of the circle
 */
double circleArea(double r) {
  const double PI = 3.1415927;     // define a constant
  return PI * r * r;               // return an expression
}

int main() {
                                   // prompt for and input radius
  cout << "To compute the area of a circle,\n enter its radius: ";
  double radius;                   // declare a variable
  cin >> radius;                   // read radius from the keyboard
                                   // compute area
  double area = circleArea(radius);
                                   // output area
  cout << "The area is " << area << endl;
}
